internal/olm/client/client.go

```diff
@@ -112,16 +112,52 @@ func NewClientForConfig(cfg *rest.Config) (*Client, error) {
 func (c Client) DoCreate(ctx context.Context, objs ...client.Object) error {
 	for _, obj := range objs {
 		kind := obj.GetObjectKind().GroupVersionKind().Kind
-		log.Infof("  Creating %s %q", kind, getName(obj.GetNamespace(), obj.GetName()))
-		err := c.KubeClient.Create(ctx, obj)
+		resourceName := getName(obj.GetNamespace(), obj.GetName())
+
+		log.Infof("  Creating %s %q", kind, resourceName)
+
+		if err := c.safeCreateOneResource(ctx, obj, kind, resourceName); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// try to create 10 times before giving up
+func (c Client) safeCreateOneResource(ctx context.Context, obj client.Object, kind string, resourceName string) error {
+	for i := 0; i < 10; i++ { // try to create 10 times before giving up
+		retry, err := c.tryCreatingOneResource(ctx, obj, kind)
 		if err != nil {
-			if !apierrors.IsAlreadyExists(err) {
-				return err
-			}
+			return err
+		}
+
+		if !retry {
+			log.Infof("    %s %q created", kind, resourceName)
+			return nil
+		}
+
+		log.Infof("    the CRD for %s %q is not ready yet. Retrying in one second", kind, resourceName)
+		time.Sleep(time.Second)
+	}
+
```

Comment:
It would be a bit more idiomatic (at least from our kubernetes adjacency) to use the client-go retry package.

See: https://pkg.go.dev/k8s.io/client-go/util/retry#OnError